source/backend/api/Areas/Leases/Controllers/ConsultationController.cs

```diff
@@ -0,0 +1,183 @@
+        public IActionResult GetLeaseConsultationById([FromRoute]long id, [FromRoute]long consultationId)
+        {
+            _logger.LogInformation(
+                "Request received by Controller: {Controller}, Action: {ControllerAction}, User: {User}, DateTime: {DateTime}",
+                nameof(ConsultationController),
+                nameof(GetLeaseConsultationById),
+                User.GetUsername(),
+                DateTime.Now);
+
+            var consultation = _leaseService.GetConsultationById(consultationId);
+
+            return new JsonResult(_mapper.Map<ConsultationLeaseModel>(consultation));
+        }
+
+        /// <summary>
+        /// Update the lease file consultation by Id.
+        /// </summary>
+        /// <returns>The consultation item updated.</returns>
+        [HttpPut("{id:long}/consultations/{consultationId:long}")]
+        [HasPermission(Permissions.LeaseEdit)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(ConsultationLeaseModel), 200)]
+        [TypeFilter(typeof(NullJsonResultFilter))]
+        [SwaggerOperation(Tags = new[] { "leasefile" })]
+        public IActionResult UpdateLeaseConsultation([FromRoute] long id, [FromRoute] long consultationId, [FromBody] ConsultationLeaseModel consultation)
+        {
+            _logger.LogInformation(
+                "Request received by Controller: {Controller}, Action: {ControllerAction}, User: {User}, DateTime: {DateTime}",
+                nameof(ConsultationController),
+                nameof(UpdateLeaseConsultation),
+                User.GetUsername(),
+                DateTime.Now);
+
+            if (id != consultation.LeaseId)
+            {
+                throw new BadRequestException("Invalid LeaseId.");
+            }
+
+            var updatedConsultation = _leaseService.UpdateConsultation(_mapper.Map<Dal.Entities.PimsLeaseConsultation>(consultation));
+
+            return new JsonResult(_mapper.Map<ConsultationLeaseModel>(updatedConsultation));
+        }
+
+        /// <summary>
+        /// Delete the lease file consultation by Id.
+        /// </summary>
+        /// <returns>The consultation item updated.</returns>
+        [HttpDelete("{id:long}/consultations/{consultationId:long}")]
+        [HasPermission(Permissions.LeaseView)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(bool), 200)]
+        [SwaggerOperation(Tags = new[] { "leasefile" })]
+        public IActionResult DeleteLeaseConsultation([FromRoute] long id, [FromRoute] long consultationId)
+        {
+            _logger.LogInformation(
+                "Request received by Controller: {Controller}, Action: {ControllerAction}, User: {User}, DateTime: {DateTime}",
+                nameof(ConsultationController),
+                nameof(DeleteLeaseConsultation),
+                User.GetUsername(),
+                DateTime.Now);
+
+            var result = _leaseService.DeleteConsultation(consultationId);
```

Comment:
not sure if it matters, but:
1) the lease id is unused.
2) the other endpoints validate that the lease id passed is valid for the given consultationId.